### DISCUSSION TITLE
How to precompiling Templates Inside NodeJS using .handlebars file

### DISCUSSION THREAD

**Discussion by namleasia** [handlebars-lang/handlebars.js#1792] [Q&A] [answered]
From the document:
```javascript
let template = "Handlebars <b>{{doesWhat}}</b> precompiled!";
let Handlebars = require("handlebars");
let compiled = Handlebars.precompile(template);
console.log(compiled);
```

The `template` is a string.
How to use a .handlebars file instead of string?

**Comment by jaylinski** [comment, marked as answer]
You can read a file via `fs`, see https://nodejs.org/en/knowledge/file-system/how-to-read-files-in-nodejs/.

**Reply by namleasia** [reply]
Thank you, but is there handlebars way to do that?

**Reply by jaylinski** [reply]
There isn't as far as I know:

https://github.com/handlebars-lang/handlebars.js/blob/c295ef085f6ae756bfb193fed6b0c48ad9a32433/lib/handlebars/compiler/compiler.js#L486-L495
